ql/ql/src/queries/style/ValidatePredicateGetReturns.ql

```diff
@@ -0,0 +1,48 @@
+/**
+ * @name Predicates starting with "get" or "as" should return a value
+ * @description Checks if predicates that start with "get" or "as" actually return a value.
+ * @kind problem
+ * @problem.severity warning
+ * @id ql/predicates-get-should-return-value
+ * @tags correctness
+ *       maintainability
+ * @precision high
+ */
+
+import ql
+import codeql_ql.ast.Ast
+
+/**
+ * Identifies predicates whose names start with "get", "as" followed by an uppercase letter.
+ * This ensures that only predicates like "getValue" are matched, excluding names like "getter".
+ */
+predicate isGetPredicate(Predicate pred) { pred.getName().regexpMatch("(get|as)[A-Z].*") }
+
+/**
+ * Checks if a predicate has a return type.
+ */
+predicate hasReturnType(Predicate pred) { exists(pred.getReturnTypeExpr()) }
+
+/**
+ * Checks if a predicate is an alias using getAlias().
+ */
+predicate isAlias(Predicate pred) { exists(pred.(ClasslessPredicate).getAlias()) }
+
+/**
+ * Returns "get" if the predicate name starts with "get", otherwise "as".
+ */
+string getPrefix(Predicate pred) {
```

Comment:
There's really no need for this predicate - just replace `regexpMatch` with `regexpCapture` in `isGetPredicate` and add the prefix column to that predicate.

Reply:
I see, that does make it cleaner!
67745e6332564b343537ab63e08330a056eeed4f